src/Sulu/Bundle/WebsiteBundle/Controller/DefaultController.php

```diff
@@ -64,4 +62,38 @@ public function redirectAction(Request $request)
     {
         return new RedirectResponse($request->get('url'), 301);
     }
+
+    /**
+     * Resolve the redirect URL, appending any additional path data
+     *
+     * @param string $url Original webspace URI
+     * @param string $redirectUrl  Redirect webspace URI
+     * @param string $requestUri The actual incoming request URI
+     *
+     * @return string URL to redirect to
+     */
+    protected function resolveRedirectUrl($portalUrl, $redirectUrl, $requestUri)
+    {
+        preg_match(
+            sprintf('{%s(:.*?)?/(.*)}', $portalUrl),
+            $requestUri,
+            $matches
+        );
+
+        $port = isset($matches[1]) ? $matches[1] : null;
+
+        $url = preg_replace(
+            sprintf('{%s(:.*?)?/(.*)}', $portalUrl),
+            sprintf('%s/\2', $redirectUrl),
+            $requestUri
+        );
+
+        if ($port) {
+            $url = preg_replace('{(http://)?(.*?)/(.*)$}', sprintf('\1\2%s/\3', $port), $url);
+        }
+
+        $url = rtrim($url, '/');
```

Comment:
We should use pathInfo or a Symfony equivalent function to split the url in Domain/Path/Query. Because only the Path should be rtrim currently the redirects dont work when had a query string. e.g.
sulu.lo/?id=12, sulu.lo?id=12, sulu.lo/articles/?id=12, sulu.lo/articles/?id=12